Core/GDCore/IDE/Events/ExpressionValidator.cpp

```diff
@@ -77,6 +77,28 @@ ExpressionValidator::Type ExpressionValidator::ValidateFunction(const gd::Functi
       MetadataProvider::GetBehaviorAnyExpressionMetadata(
             platform, behaviorType, function.functionName);
 
+  Type returnType = StringToType(metadata.GetReturnType());
+
+  if (!function.objectName.empty() &&
```

Comment:
you can probably wrap the 3 ifs doing a check on `!function.objectName.empty()` in to one to avoid making the call 3 times

Reply:
I only see 2 of them. The one in the middle is `!function.behaviorName.empty()`. So I don't see how to simplify them.
The 3rd if has a useless nested if, I merged it.

Reply:
oh yes misread it. nice 👍